dbt/include/trino/macros/adapters.sql

```diff
@@ -173,8 +173,16 @@
     {% set rel_identifier = row[1] %}
     {% set rel_schema = row[2] %}
     {% set rel_type = api.Relation.get_relation_type(row[3]) %}
-    {% set existing = api.Relation.create(database=rel_db, schema=rel_schema, identifier=rel_identifier, type=rel_type) %}
-    {% do drop_relation(existing) %}
+    {# --Do not drop Iceberg storage table, when dropping Iceberg materialized view #}
+    {% if not (rel_db == 'iceberg'
+      and rel_type == 'table'
+      and rel_identifier.startswith('st_')
+      and rel_identifier.count('_') == 1
+      and rel_identifier|length == 35)
+    %}
+        {% set existing = api.Relation.create(database=rel_db, schema=rel_schema, identifier=rel_identifier, type=rel_type) %}
+        {% do drop_relation(existing) %}
+    {% endif %}
```

Comment:
Please revert this change. This is an adapter method, we can't adapt adapter code to make a test (cleanup) work. We should just adapt the config for the storage table to store to another schema and drop that schema manually.

Check the [trino docs](https://trino.io/docs/current/connector/iceberg.html#general-configuration):

`iceberg.materialized-views.storage-schema` | Schema for creating materialized views storage tables. When this property is not configured, storage tables are created in the same schema as the materialized view definition. When the storage_schema materialized view property is specified, it takes precedence over this catalog property.
-- | --

Reply:
But this is not only about tests, but about general usage of this macro. If we or users would create some custom macro and use `drop_schema` there, it would throw error if they would try to drop schema with materialized views. I think it should work straight away, without need of change any config to be able to use this macro.

This macro is already somewhat adapted to workaround lack of support for CASCADE. But now, it is [supported in Trino](https://trino.io/docs/current/sql/drop-schema.html). After adding support for CASCADE in great lakes connectors ([issue for that](https://github.com/trinodb/trino/issues/17649)), maybe `trino__drop_schema` macro could be completely removed to use [dbt-core default implementation](https://github.com/dbt-labs/dbt-core/blob/fe9c875d32554cf36a08a94a6ee5d356066d80f8/core/dbt/include/global_project/macros/adapters/schema.sql#L16-L20) instead. 

There is not elegant hardcoding in proposed workaround, but it should work for every case on Iceberg (every Iceberg storage table name starts with _st__ and consists of 35 characters). Maybe then it would be better to adapt this macro so it would work for every usage of it, rather than adjusting test case?
WDYT?

Reply:
`rel_db` is not the type of the catalog but just the name of the catalog, so I don't really see the value of the proposed solution.

As this is a fairly new feature in Trino I think it is a bit too early to switch to `CASCADE` syntax.

I also think no user raised this issue, so why try to tackle it in adapter code. Most of the time users will actually configure the storage schema to be different from the user schema (as that makes it easier to manage from a security perspective, the storage schema is rather internal to the database engine).

Reply:
Fair enough, I overlooked that 
> rel_db is not the type of the catalog but just the name of the catalog

I reverted changes from `trino__drop_schema` macro.

I added additional code for dropping MVs before dropping schema, so we are no longer facing this issue.